### PR TITLE
Feature/319 implement basic log collector

### LINKED ISSUE #319: Implement Basic LogCollector
## User stories
As a datascience developer I want to use the log collector in order to accumulate all data that was generate by the logger.

## Dependencies
Depends on: 
 - #16 

## (Wiki) References
- https://github.com/BP2022-AP1/bp2022-ap1/wiki/Architecture#logger
- https://github.com/BP2022-AP1/bp2022-ap1/wiki/Logging

## Acceptance Criteria
- [x] able to request all departures and arrivals of all trains from logs by run_id
- [x] able to request leave/enter times for all trains of a run_id with element_id & length

### PR DESCRIPTION
**Pull request by LucasDerReisende**
Fixes #319 

Implements the first basic log collector interface to generate pandas dataframes from log entries in database

## PR checklist

- [x] Acceptance criteria fulfilled
- [x] Additional features are tested
- [x] Docs (code, wiki & diagrams) updated
- [x] Breaking changes anounced
- [x] Added classes that inherit from `BaseModel` to `src.constants.tables`
- [x] Dev-branch has been merged into local branch to resolve conflicts
- [x] Tests and linter have passed AFTER local merge
- [x] Another dev reviewed and approved

### CODE REVIEW

File: src/logger/log_collector.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+                    and TrainLeaveBlockSectionLogEntry.train_id == train_id
+                )
+            ],
+            columns=["tick", "block_section_id", "block_section_length"],
+        )
+        train_enter_df = train_enter_df.sort_values("tick")
+        train_leave_df = train_leave_df.sort_values("tick")
+
+        train_enter_first = train_enter_df.iloc[0].tick < train_leave_df.iloc[0].tick
+        train_enter_last = train_enter_df.iloc[-1].tick >= train_leave_df.iloc[-1].tick
```
Comment (instant-sky) [inline src/logger/log_collector.py:189]: in `get_departures_arrivals_of_train` you just used `>` compared to the `>=` here. Does this make a difference?

File: tests/logger/test_log_collector.py [hunk 2]
```diff
@@ -0,0 +1,306 @@
+import pandas as pd
+import pytest
+from pandas._testing import assert_frame_equal
```
Comment (instant-sky) [inline tests/logger/test_log_collector.py]: Is this the "official"/best way?

**Review by instant-sky — COMMENTED** [review]
Very interesting new features!

File: tests/logger/test_log_collector.py (hunk 2 above)
Comment (LucasDerReisende) [reply tests/logger/test_log_collector.py]: I changed it :)

**Review by instant-sky — APPROVED** [review]
Looks good now!

**Pull request merged**